addons/popochiu/engine/objects/region/popochiu_region.gd

```diff
@@ -63,28 +63,31 @@ func _set_enabled(value: bool) -> void:
 #endregion
 
 #region Private ####################################################################################
-func _check_area(area: PopochiuCharacter, entered: bool) -> void:
-	if area is PopochiuCharacter:
-		if entered:
-			_on_character_entered(area)
-		else:
-			_on_character_exited(area)
+func _check_area(area: Area2D, entered: bool) -> void:
+	if not area is PopochiuCharacter: return
+	
+	if entered:
+		_on_character_entered(area)
+	else:
+		_on_character_exited(area)
 
 
 func _check_scaling(
 	area_rid: RID, area: Area2D, area_shape_index: int, local_shape_index: int, entered: bool
 ):
-	if (
+	if not (
 		area is PopochiuCharacter 
 		and area.get_node_or_null("ScalingPolygon") 
 		and area_shape_index == area.get_node("ScalingPolygon").get_index()
-		):
-		if entered:
-			if scaling:
-				_update_scaling_region(area)
-				R.current.update_character_scale(area)
-		else:
-			_clear_scaling_region(area)
+	):
+		return
+	
+	if entered:
+		if scaling:
+			_update_scaling_region(area)
+			R.current.update_character_scale(area)
+	else:
+		_clear_scaling_region(area)
```

Comment:
```suggestion
	if not entered:
		_clear_scaling_region(area)
		return
		
	if scaling:
		_update_scaling_region(area)
		R.current.update_character_scale(area)
```

Reply:
Much better. I'm gonna commit your suggestion tomorrow (because I want to commit all your suggestions once I finish reading the whole thing).